### PR TITLE
Extended data binding sample with checkbox and select control binding

### PR DESCRIPTION
**Pull request by C-Core**
* Added a sample how to do two-way-binding of a checkbox to a bool variable.
* Added a sample how to fill a select control dynamically using data binding and also bind the selected index to an int variable.

### CODE REVIEW

File: Include/RmlUi/Core/Elements/ElementFormControlSelect.h [hunk 1]
```diff
@@ -46,6 +46,8 @@ class WidgetDropDown;
 class RMLUICORE_API ElementFormControlSelect : public ElementFormControl
 {
 public:
+	RMLUI_RTTI_DefineWithParent(ElementFormControlSelect, ElementFormControl)
```
Comment (C-Core) [inline Include/RmlUi/Core/Elements/ElementFormControlSelect.h:49]: This has nothing to do with the data binding but I stumbled across this missing rtti some time ago.

File: Samples/basic/databinding/data/databinding.rml [hunk 2]
```diff
@@ -215,22 +215,20 @@ form h2 {
...
 		</div>
 		<h2>Subject</h2>
 		<div>
-			<select name="subject">
-				<option value="none" selected>Choose your subject</option>
-				<option value="feature">Feature request</option>
-				<option value="bug">Bug report</option>
-				<option value="praise">Praise</option>
-				<option value="criticism">Criticism</option>
+			<select name="subject" data-event-change="selected_subject = ev.value">
```
Comment (C-Core) [inline Samples/basic/databinding/data/databinding.rml:228]: @mikke89 This doesn't work unfortunately: 
Could not find variable name 'it_index' in data model.
Error in data expression at 9. Could not find data variable with name 'it_index'.
  "it_index == selected_subject"

File: Samples/basic/databinding/data/databinding.rml [hunk 3]
```diff
@@ -215,22 +215,20 @@ form h2 {
...
+				<option data-for="s : subjects" data-value="it_index" data-attrif-selected="it_index == selected_subject">{{s}}</option>
```
Comment (C-Core) [inline Samples/basic/databinding/data/databinding.rml:229]: @mikke89 This doesn't work unfortunately:
Could not find variable name 'it_index' in data model.
Error in data expression at 9. Could not find data variable with name 'it_index'.
"it_index == selected_subject"

File: Samples/basic/databinding/data/databinding.rml [hunk 4]
```diff
@@ -215,22 +215,20 @@ form h2 {
 		<div>
 			<input type="checkbox" name="meals" value="pizza" data-attrif-checked="rating > 70" disabled/> Pizza
 			<input type="checkbox" name="meals" value="pasta" checked/> Pasta
-			<input type="checkbox" name="meals" value="lasagne" checked/> Lasagne
+			<input type="checkbox" name="meals" value="lasagne" data-attrif-checked="lasagne" data-event-change="lasagne = ev.value != ''"/> Lasagne
```
Comment (C-Core) [inline Samples/basic/databinding/data/databinding.rml:218]: @mikke89 It almost worked as you suggested, but I needed to extent the expression with != '' to convert the string value into a bool. The way you declare a checkbox and its state doesn't work very intuitive with the data binding, I know this design comes from html, so I guess there is not much we can do to make this binding simpler?

**Comment by mikke89** [comment]
Thanks a lot! This is great, and it also exposes some quirks with the data binding that we should resolve.

This approach to handle checkboxes is certainly something we need to improve. I also played a bit with radio buttons and that is even harder. I think we should introduce another data view/controller to handle these cases more gracefully, but I'm not exactly sure what it should look like.

As for the `it_index` not being found. Yes I think right now this only works for children. I'll try to make it work on the same element.

I'll merge this into the `data_binding` branch for now to work on it some more. Thanks again!

**Comment by C-Core** [comment]
@mikke89 I also found that data-attrif-selected doesn't work for option elements. I debugged a bit through the code, it does add or remove the attribute correctly, but the selected entry doesn't change in the UI.

**Comment by mikke89** [comment]
In that case, I think it's better to set the `value` attribute on the parent `select` element.

**Comment by C-Core** [comment]
Ok, I can bind a variable to the select control with data-value. But if the variable is changed in code or by another control it still doesn't change the selected option for the select control.

**Comment by mikke89** [comment]
I see, thanks for testing. I'll add it to the todo-list :)

**Pull request merged**